### PR TITLE
Enh #19884: Added support Enums in Query Builder

### PR DESCRIPTION
**Pull request by sk1t0n**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  |  #19884

### CODE REVIEW

File: framework/CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Yii Framework 2 Change Log
 - Enh #19841: Allow jQuery 3.7 to be installed (wouter90)
 - Enh #19853: Added support for default value for `\yii\helpers\Console::select()` (rhertogh)
 - Bug #19868: Added whitespace sanitation for tests, due to updates in ICU 72 (schmunk42)
-
```
Comment (rhertogh) [inline framework/CHANGELOG.md:11]: New line is missing (there should be 2 empty lines after the last text)

File: framework/db/Command.php [hunk 2]
```diff
@@ -377,6 +377,11 @@ public function bindValues($values)
                 $this->pendingParams[$name] = [$value->getValue(), $value->getType()];
                 $this->params[$name] = $value->getValue();
             } else {
+                if (version_compare(PHP_VERSION, '8.1.0') >= 0
+                    && $value instanceof \UnitEnum
+                ) {
+                    $value = isset($value->value) ? $value->value : $value->name;
```
Comment (rhertogh) [inline framework/db/Command.php:383]: Could we use '$value instanceof  \BackedEnum` here?

File: tests/framework/db/CommandTest.php [hunk 3]
```diff
@@ -1525,4 +1525,21 @@ public function testBindValuesSupportsDeprecatedPDOCastingFormat()
         $db->createCommand()->setSql("SELECT :p1")->bindValues([':p1' => [2, \PDO::PARAM_STR]]);
         $this->assertTrue(true);
     }
+    
+    public function testBindValuesSupportsEnums()
+	{
+		if (version_compare(PHP_VERSION, '8.1.0') >= 0) {
```
Comment (rhertogh) [inline tests/framework/db/CommandTest.php:1531]: You can mark a test as "skipped", please see https://docs.phpunit.de/en/9.6/incomplete-and-skipped-tests.html#skipping-tests

```php
if (version_compare(PHP_VERSION, '8.1.0', '<')) {
   $this->markTestSkipped('Enums are not supported in PHP < 8.1');
}
```

**Review by rhertogh — CHANGES_REQUESTED** [review]

File: framework/db/Command.php (hunk 2 above)
Comment (sk1t0n) [reply framework/db/Command.php:383]: This is a universal solution so you don't have to check for BackedEnum separately.

```php
$command->setSql('SELECT :p1')->bindValues([':p1' => enums\Status::ACTIVE]);
$this->assertSame('ACTIVE', $command->params[':p1']);

$command->setSql('SELECT :p1')->bindValues([':p1' => enums\StatusTypeString::ACTIVE]);
$this->assertSame('active', $command->params[':p1']);

$command->setSql('SELECT :p1')->bindValues([':p1' => enums\StatusTypeInt::ACTIVE]);
$this->assertSame(1, $command->params[':p1']);
```

Comment (rhertogh) [reply framework/db/Command.php:383]: I've been a bit unclear. What I meant was to replace the `isset` check:
`$value = $value instanceof \BackedEnum ? $value->value : $value->name;`

File: tests/framework/db/CommandTest.php (hunk 3 above)
Comment (sk1t0n) [reply tests/framework/db/CommandTest.php:1531]: OK. Thank you.

File: framework/CHANGELOG.md (hunk 1 above)
Comment (sk1t0n) [reply framework/CHANGELOG.md:11]: Added a correction.

File: framework/db/Command.php (hunk 2 above)
Comment (sk1t0n) [reply framework/db/Command.php:383]: What do you think of this solution?

```php
if (version_compare(PHP_VERSION, '8.1.0') >= 0) {
    if ($value instanceof \BackedEnum) {
        $value = $value->value;
    } elseif ($value instanceof \UnitEnum) {
        $value = $value->name;
    }
}
````

Comment (sk1t0n) [reply framework/db/Command.php]: Added a correction.

**Review by rhertogh — APPROVED** [review]

**Review by Arhell — APPROVED** [review]

**Review by bizley — APPROVED** [review]

**Comment by samdark** [comment]
Thanks!

**Pull request merged**